internal/provwasm/stargate_whitelist.go

```diff
@@ -0,0 +1,163 @@
+package provwasm
+
+import (
+	"fmt"
+	"sync"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	ibctransfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	attributetypes "github.com/provenance-io/provenance/x/attribute/types"
+	markertypes "github.com/provenance-io/provenance/x/marker/types"
+	metadatatypes "github.com/provenance-io/provenance/x/metadata/types"
+	msgfeestypes "github.com/provenance-io/provenance/x/msgfees/types"
+	nametypes "github.com/provenance-io/provenance/x/name/types"
+	rewardtypes "github.com/provenance-io/provenance/x/reward/types"
+)
+
+// stargateWhitelist keeps whitelist and its deterministic
+// response binding for stargate queries.
+//
+// The query can be multi-thread, so we have to use
+// thread safe sync.Map.
+var stargateWhitelist sync.Map
+
+// Note: When adding a migration here, we should also add it to the Async ICQ params in the upgrade.
+// In the future we may want to find a better way to keep these in sync
+
+//nolint:staticcheck
+func init() {
+	// ibc queries
+	setWhitelistedQuery("/ibc.applications.transfer.v1.Query/DenomTrace", &ibctransfertypes.QueryDenomTraceResponse{})
+
+	// ==========================================================
+	// cosmos-sdk queries
+	// ==========================================================
+
+	// auth
+	setWhitelistedQuery("/cosmos.auth.v1beta1.Query/Account", &authtypes.QueryAccountResponse{})
```

Comment:
Are all the queries deterministic? We might want to double check.

Reply:
we should be able to add all those from the Osmosis project right away...
https://github.com/osmosis-labs/osmosis/blob/main/wasmbinding/stargate_whitelist.go

Reply:
The metadata ones are all deterministic. We might consider omitting the `__All` queries in there though since they can easily time out for large counts.

Reply:
I glanced through all the other query server keeper files and didn't see anything that looked non-deterministic. Whenever we are returning lists, we're either using the SDK's pagination stuff or iterating over keys directly.

The only one that caught my eye was the msgfees `CalculateTxFees` query. It simulates a tx to identify cost, and I don't know how deterministic that is. My main thought is about something being in our out of a cache.

Reply:
yeah i've left `CalculateTxFees` out of this side and provwasm since it's got a float and seemed lower priority for this release. otherwise, i'd need to add transformation logic in provwasm to work around that

Reply:
i'll remove the `_All` queries

Reply:
Feels like we should remove the _All queries as they don't seem like the kind of thing we would want people to use from within a smart contract...  And the bigger ones on metadata, markers, etc are definitely going to time out.  I could see the name module ones being a bit better though.  Even there some people have started adding tons of uuid/address entries...

Reply:
alright, removed those queries and the `AttributeAccounts` since that could also be quite large